### PR TITLE
Fix AI-VAD issues

### PR DESCRIPTION
**Pull request by djdameln**
## Description

This PR fixes the following issues with the AI-VAD model implementation:
- Error when zero bounding boxes were detected in a video frame.
- Problems with model saving and loading due to use of non-pytorch modules in density estimators.

To address the latter issue, a pytorch-only implementation of Gaussian Mixture Models was added.

Also sets the default `val_split_mode` of the video datasets to `SAME_AS_TEST` to enforce determistic behaviour between runs and to be in line with the image datasets.

## Known Issues

Export is not working because the current design of the export functionality does not support video models.

## Changes

<details>
<summary>Describe the changes you made</summary>

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks @djdameln. Overall, looking great. I've got some comments in regards to the documentation and annotation stuff.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks! Really clean code.

**Review by samet-akcay — APPROVED** [review]
Thanks a lot!

File: src/anomalib/models/components/cluster/gmm.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+        self.means: Tensor
+        self.covariances: Tensor
+        self.weights: Tensor
+
+    def fit(self, data: Tensor) -> None:
+        """Fit the model to the data.
+
+        Args:
+            data (Tensor): Data to fit the model to. Tensor of shape (n_samples, n_features).
```
[the remark this answers is not in the document]
Comment (djdameln) [reply src/anomalib/models/components/cluster/gmm.py:71]: The main docstring of the module now covers all the entrypoint methods. Would this be sufficient? Otherwise there will be a lot of duplication because each example involves generating some data and fitting a gmm.

File: src/anomalib/data/video/avenue.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def __init__(
         num_workers: int = 8,
         transform_config_train: str | A.Compose | None = None,
         transform_config_eval: str | A.Compose | None = None,
-        val_split_mode: ValSplitMode = ValSplitMode.FROM_TEST,
+        val_split_mode: ValSplitMode = ValSplitMode.SAME_AS_TEST,
```
[the remark this answers is not in the document]
Comment (djdameln) [reply src/anomalib/data/video/avenue.py:215]: All image datasets except `Folder` already have this as the default behaviour. The motivation for having FROM_TEST in the Folder dataset is that `FROM_TEST` is actually a better practice, because it leads to mutually exclusive validation and test sets. The reason we use `SAME_AS_TEST` for the other datasets is for reproducibility: Results in literature are evaluated over the entire test set from which no samples are withheld for validation. Because Folder dataset always concerns custom data, we can safely create a val-test split.

I do feel that maybe consistent behaviour across all datasets in desired in this case, so I could change the behaviour of Folder to `SAME_AS_TEST` in a separate PR. What do you think? @samet-akcay @ashwinvaidya17

[the remark this answers is not in the document]
Comment (samet-akcay) [reply src/anomalib/data/video/avenue.py:215]: hm, if the other benchmarks is correct, then we could leave it for now. I feel that we might need to revisit folder dataset in the future anyway. We currently do not address if the user already has a validation set
